src/CDN/CloudFront.php

```diff
@@ -77,17 +78,50 @@ class CloudFront implements CDNInterface
     protected $client;
 
     /**
-     * @param string $path
-     * @param string $key
-     * @param string $secret
-     * @param string $distributionId
+     * @var string
      */
-    public function __construct($path, $key, $secret, $distributionId)
-    {
+    private $region;
+
+    /**
+     * @var string
+     */
+    private $version;
+
+    /**
+     * @todo: Make mandatory argument 5 and 6 when support for aws/aws-sdk-php < 3.0 is dropped.
+     */
+    public function __construct(
+        string $path,
```

Comment:
This class is not final, adding type hints is a BC break, also this class won't work with invalid parameters

Reply:
Comment addressed.